### PR TITLE
feat: Remove n from SelectedUnitsCounts{,Sorted}

### PR DESCRIPTION
**Pull request by badosu**
Fixes https://github.com/beyond-all-reason/spring/issues/669

### CODE REVIEW

File: doc/site/migrating-from-spring.markdown [hunk 1]
```diff
@@ -6,43 +6,88 @@ nav_order: 5
...
-will not be the same color as previously. To get back previous behaviour,
-you might need to add
+## Callouts
+
+### Font rendering
+
+Rendering fonts now obeys GL color state. This means that sometimes text will
+not be the same color as previously. To get back previous behaviour, you might
+need to add
 [`gl.Color`](https://beyond-all-reason.github.io/spring/ldoc/modules/OpenGL.html#gl.Text)
```
Comment (badosu) [inline doc/site/migrating-from-spring.markdown:33]: Shouldn't we recommend using lua fonts instead given AtiHacks issues?

File: doc/site/migrating-from-spring.markdown [hunk 2]
```diff
@@ -6,43 +6,88 @@ nav_order: 5
...
-There is no way to get back the old behaviour for now,
-but if someone needs it it could be arranged.
-- Rendering fonts now obeys GL color state. This means that sometimes text
-will not be the same color as previously. To get back previous behaviour,
-you might need to add
+## Callouts
+
+### Font rendering
+
+Rendering fonts now obeys GL color state. This means that sometimes text will
```
Comment (sprunk) [inline doc/site/migrating-from-spring.markdown:30]: looks like some unrelated changes got in?

Comment (badosu) [reply doc/site/migrating-from-spring.markdown:30]: They are a restructuring of the file. You can look at the addition of this change as a 'breaking' change on the same file.

File: doc/site/migrating-from-spring.markdown [hunk 3]
```diff
@@ -6,43 +6,88 @@ nav_order: 5
...
+
+The tables returned by
+[`Spring.GetSelectedUnitsSorted`](https://beyond-all-reason.github.io/spring/ldoc/modules/UnsyncedRead.html#Spring.GetSelectedUnitsSorted) and
+[`Spring.GetSelectedUnitsCounts`](https://beyond-all-reason.github.io/spring/ldoc/modules/UnsyncedRead.html#Spring.GetSelectedUnitsCounts)
+no longer have an additional `n` key with corresponding value containining the
+number of unitdefs.
+
+Instead use the second return value, e.g.:
+
+```diff
```
Comment (badosu) [inline doc/site/migrating-from-spring.markdown:85]: @sprunk

Comment (sprunk) [reply doc/site/migrating-from-spring.markdown:85]: Looks good

**Review by sprunk — APPROVED** [review]

**Comment by badosu** [comment]
Tested, ready for review

**Review by sprunk — APPROVED** [review]
+1 for the `n` part. I don't see why you would bundle the markdown rewrite into this PR rather than do it separately (and just add the few lines related to the `n` change at the bottom) but I don't personally mind.

**Comment by badosu** [comment]
> I don't see why you would bundle the markdown rewrite into this PR rather than do it separately

I'm avoiding pushing too many separate commits for stuff that doesnt really touch cpp code.

**Comment by sprunk** [comment]
Sure, makes sense.

**Comment by badosu** [comment]
I don't mind it, but does it make sense to document breaking changes on two places? The markdown document is just as readable from an editor.

**Comment by sprunk** [comment]
I think ideally the whole changelog would be in the same place (with breaking changes somehow marked and made more prominent), and the markdown doc currently doesn't contain non-breaking changes. I think I agree that the markdown doc would be superior, but as-is the .txt is both more complete and is where I'd assume the 105 remnants would be looking because that's how it has always been.
